src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBaker.java

```diff
@@ -76,7 +126,207 @@ public String getJobName()
     @Override
     public AbstractJob createJob(final CitizenData citizen)
     {
-        return new JobPlaceholder(citizen); //TODO Implement Later
+        return new JobBaker(citizen);
+    }
+
+    /**
+     * Clear the furnaces list.
+     */
+    public void clearFurnaces()
+    {
+        furnaces.clear();
+    }
+
+    /**
+     * Add a furnace to the building.
+     *
+     * @param pos the position of it.
+     */
+    public void addToFurnaces(final BlockPos pos)
+    {
+        furnaces.put(pos, null);
+    }
+
+    /**
+     * Remove a furnace from the building.
+     *
+     * @param pos the position of it.
+     */
+    public void removeFromFurnaces(final BlockPos pos)
+    {
+        furnaces.remove(pos);
+    }
+
+    /**
+     * Remove a product from the furnace.
+     * @param pos the position the furnace is at.
+     */
+    public void removeProductFromFurnace(final BlockPos pos)
+    {
+        furnaces.replace(pos, null);
+    }
+
+    /**
+     * Return a list of furnaces assigned to this hut.
+     *
+     * @return copy of the list
+     */
+    public List<BlockPos> getFurnaces()
+    {
+        return new ArrayList<>(furnaces.keySet());
+    }
+
+    /**
+     * Return the map of furnaces assigned to this hut and the product in it.
+     *
+     * @return a hashmap with BlockPos, Product.
+     */
+    public Map<BlockPos, Product> getFurnacesWithProduct()
+    {
+        return new HashMap<>(furnaces);
+    }
+
+    /**
+     * Get the map of current tasks in the baker.
+     *
+     * @return the map of states and products.
+     */
+    public Map<ProductState, List<Product>> getTasks()
```

Comment:
Please use ImmutableMaps here. There is no valid or good reason for somebody else to change the Tasks a Baker has.